### PR TITLE
Fix bad bug in merging pysbd segments

### PR DESCRIPTION
**Pull request by danielkingai2**
Thanks for finding @isabelcachola, hopefully there aren't any more :)

Edit: This pr has become a couple of hacks to fall back on the default sentence splitter if pysbd does something weird or fails. This is the first stage of testing out this pipe as the sentence splitter.

Edit Edit: known pysbd issues have been fixed, so i've removed the fall backs

### CODE REVIEW

File: scispacy/custom_sentence_segmenter.py [hunk 1]
```diff
@@ -35,6 +40,7 @@ def combined_rule_sentence_segmenter(doc: Doc) -> Doc:
     current_segment = segments[segment_index]
     built_up_sentence = ""
     for i, token in enumerate(doc):
+        # print(current_segment, "BREAK", built_up_sentence, "BREAK", token)
```
Comment (DeNeutoy) [inline scispacy/custom_sentence_segmenter.py]: remove commented code

File: scispacy/custom_sentence_segmenter.py [hunk 2]
```diff
@@ -26,7 +31,16 @@ def combined_rule_sentence_segmenter(doc: Doc) -> Doc:
 
     @param doc: the spaCy document to be annotated with sentence boundaries
     """
-    segmenter = pysbd.Segmenter(language="en", clean=False)
+    # pysbd doesn't always like short strings
```
Comment (nipunsadvilkar) [inline scispacy/custom_sentence_segmenter.py]: @danielkingai2 : Can you share few examples of such cases?

Comment (danielkingai2) [reply scispacy/custom_sentence_segmenter.py]: @nipunsadvilkar the one i noticed was that it turned `.` into ``. And it seemed harmless to not split segments on one single token documents so I just special cased it.

Comment (nipunsadvilkar) [reply scispacy/custom_sentence_segmenter.py]: I didnt understand it fully. Can you provide an example, please?

Comment (danielkingai2) [reply scispacy/custom_sentence_segmenter.py]:
```
>>> import pysbd
>>> segmenter = pysbd.Segmenter(language='en', clean=False)
>>> segmenter.segment('.')
[]
>>> 
```

Comment (nipunsadvilkar) [reply scispacy/custom_sentence_segmenter.py]: Got it. Thanks 👍

Comment (nipunsadvilkar) [reply scispacy/custom_sentence_segmenter.py]: bdw I have fixed https://github.com/nipunsadvilkar/pySBD/issues/27 bug.

**Comment by danielkingai2** [comment]
Current status is that all tests pass without the fall backs on the normal sentence splitter. Leaving them in for now (without merging), hoping that @isabelcachola can rerun her job and see if there are any more edge cases that output warnings, or if we can remove the fall backs.

**Comment by danielkingai2** [comment]
@DeNeutoy all the hacky fallbacks have been removed because all the known pysbd bugs were fixed. I'll merge it in this (better) state once CI passes unless you object to something
